tests/integration/helpers.py

```diff
@@ -0,0 +1,29 @@
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import logging
+from typing import Optional
+
+from pytest_operator.plugin import OpsTest
+
+logger = logging.getLogger(__name__)
+
+
+async def get_address(ops_test: OpsTest, app_name: str, unit_num: Optional[int] = None) -> str:
+    """Find unit address for any application.
+
+    Args:
+        ops_test: pytest-operator plugin
+        app_name: string name of application
+        unit_num: integer number of a juju unit
+
+    Returns:
+        unit address as a string
+    """
+    status = await ops_test.model.get_status()
+    app = status["applications"][app_name]
+    return (
+        app.public_address
+        if unit_num is None
+        else app["units"][f"{app_name}/{unit_num}"]["address"]
+    )
```

Comment:
i wonder @mateoflorido if this would be a nice addition to pylibjuju?.  Not that i'm asking you to take it out of this PR

Reply:
I think so. The O11y team is using the same helper function in the Traefik charm. Nevertheless, I believe the get_public_address method should handle this. However, it's returning None in the latest pylibjuju. There was an issue opened a while ago, so I wonder if it would be worth creating another one to raise concerns about this issue.